crates/sncast/src/lib.rs

```diff
@@ -278,22 +279,29 @@ pub async fn wait_for_tx(
     }
     let retries = timeout / u16::from(retry_interval);
     for i in (1..retries).rev() {
-        match provider.get_transaction_receipt(tx_hash).await {
-            Ok(receipt) => {
-                return match receipt.execution_result() {
-                    ExecutionResult::Succeeded => Ok("Transaction accepted"),
-                    ExecutionResult::Reverted { reason } => {
-                        Err(anyhow!("Transaction has been reverted = {reason}"))
+        match provider.get_transaction_status(tx_hash).await {
+            Ok(starknet::core::types::TransactionStatus::Rejected) => {
+                return Err(anyhow!("Transaction has been rejected"));
+            }
+            Ok(starknet::core::types::TransactionStatus::AcceptedOnL2(execution_status)
+            | starknet::core::types::TransactionStatus::AcceptedOnL1(execution_status)) => {
+                match execution_status {
+                    starknet::core::types::TransactionExecutionStatus::Succeeded => {
+                        return Ok("Transaction accepted")
+                    }
+                    starknet::core::types::TransactionExecutionStatus::Reverted => {
+                        return get_revert_reason(provider, tx_hash).await
                     }
                 }
             }
-            Err(StarknetError(TransactionHashNotFound)) => {
-                let remaining_time = i * u16::from(retry_interval);
-                println!("Waiting for transaction to be accepted ({i} retries / {remaining_time}s left until timeout)");
-            }
+            Ok(starknet::core::types::TransactionStatus::Received)
+            | Err(StarknetError(TransactionHashNotFound)) => {}
```

Comment:
Any reason why the remaining time calculation and printing cannot be included in case where received | notfound?